src/client.rs

```diff
@@ -0,0 +1,99 @@
+use reqwest::{
+    header::{HeaderMap,AUTHORIZATION, CONTENT_TYPE},
+};
+
+use crate::docker::DockerClient;
+
+/// Struct to connect define parameters to connect to Crux
+/// `host` and `port` are reuired.
+pub struct Crux {
+    host: String,
+    port: String,
+    headers: HeaderMap
+}
+
+impl Crux{
+    /// Define Crux instance with `host:port`
+    pub fn new(host: &str, port: &str) -> Self {
+        Self{host: host.to_string(), port: port.to_string(), headers: HeaderMap::new()}
+    }
+
+    /// Function to add `AUTHORIZATION` token to the Crux Client
+    pub fn with_authorization(mut self, authorization: &str) -> Self {
+        self.headers.insert(AUTHORIZATION, authorization.parse().unwrap());
+        self
+    }
+
+    #[cfg(not(test))]
+    fn uri(&self) -> String {
+        format!("http://{}:{}", self.host, self.port)
+    }
+
+    #[cfg(test)]
+    fn uri(&self) -> String {
+        use mockito::server_url;
+        server_url()
+    }
+
+    /// To query database on Docker via http it is necessary to use `DockerClient` 
+    pub fn docker_client(&mut self) -> DockerClient {
```

Comment:
Thinking now, knowing that Crux has multiple ways of being consumed, this `_client` method really become handy and it seems like a smart design decision 👏 

You can dismiss this, I just wanted to say your idea is pretty good 😊